src/lib/libast/tests/path/pathstat.c

```diff
@@ -0,0 +1,37 @@
+#include "config_ast.h"  // IWYU pragma: keep
+
+#include <stdio.h>
+#include <string.h>
+
+#include "ast.h"
+#include "terror.h"
+
+const char *link_to_temp_file = "link_to_temp_file";
+
+void cleanup_handler(void) {
+    unlink(link_to_temp_file);
+}
+
+tmain() {
+    UNUSED(argc);
+    UNUSED(argv);
+
+    const char* temp_file = "temp_file";
+
+    struct stat statbuf;
+
+    if (open(temp_file, O_CREAT, 0666) < 0) terror("Failed to create temp file");
+
+    if (pathstat(temp_file,  &statbuf) && statbuf.st_mode == 0666) terror("Failed to stat() %s", temp_file);
+
+    if (symlink(temp_file, link_to_temp_file) < 0) terror("Failed to create symbolic link to %s", temp_file);
+
+    atexit(cleanup_handler);
```

Comment:
We should arrange to run these tests in a unique temp dir like we do for the ksh tests. Thus obviating the need for these kind of cleanup functions.